src/core/components/hv-route/index.tsx

```diff
@@ -362,8 +362,17 @@ export default function HvRoute(props: Types.Props) {
   const id: string | undefined =
     props.route?.params?.id || navigatorContext.initialRouteName || undefined;
 
+  const state = props.navigation?.getState();
+  const index = state?.index;
+  const type = state?.type;
```

Comment:
minor/more concise way to write this:
```suggestion
  const { index, type } = props.navigation?.getState() || {};
```